docs/sphinx/developer/concepts/kernel.rst

```diff
@@ -8,7 +8,36 @@ The SimEng Kernel is split into two classes, ``LinuxProcess`` and ``Linux``.
 LinuxProcess
 ------------
 
-The ``LinuxProcess`` class provides the functionality to process the supplied program. It creates the initial process memory space, including the Executable and Linkable Format (ELF) process image and the stack. The population of the initial stack state is based on the information `here <https://www.win.tue.nl/~aeb/linux/hh/stack-layout.html>`_.
+The ``LinuxProcess`` class provides the functionality to process the supplied program. It creates the initial process memory space, including the Executable and Linkable Format (ELF) process image and the stack.
+The process memory space created contains all date required by the program to run.
+
+ELF Parsing
+~~~~~~~~~~~~
+The ELF binaries have a defined structure for 32-bit and 64-bit architectures, all information regarding parsing ELF binaries has been refereced from the `Linux manual page <https://man7.org/linux/man-pages/man5/elf.5.html>`_.
+The ELF binary is divided into multiple parts. SimEng stores all relevant parts of the `ELF Binary` in a ``char[] processImage`` array, which is private member variable of the ``LinuxProcess`` class.
+
+ **ELF Header**
+
+ This is First part of ELF binary is the `ELF Header`, for 64-bit architecture the ``elf64_hdr`` struct in the Linux source tree defines the ELF Header. The first member of struct ``unsigned char e_indent`` holds information related to interpreting the ELF binary
```

Comment:
"This is the First part of the ELF binary is the `ELF Header`, for 64-bit architecture the ..." -> "The first part of the ELF binary is the `ELF Header`. For 64-bit architectures the ..."